contextualized/modules.py

```diff
@@ -45,8 +45,11 @@ class MLP(nn.Module):
     """
     def __init__(self, input_dim, output_dim, width, layers, activation=nn.ReLU, link_fn=identity_link):
         super(MLP, self).__init__()
-        hidden_layers = lambda: [layer for _ in range(0, layers - 2) for layer in (nn.Linear(width, width), activation())]
-        mlp_layers = [nn.Linear(input_dim, width), activation()] + hidden_layers() + [nn.Linear(width, output_dim)]
+        if layers > 0:
+            hidden_layers = lambda: [layer for _ in range(0, layers - 1) for layer in (nn.Linear(width, width), activation())]
```

Comment:
Is `hidden_layers` used elsewhere? Seems a bit confusing to define a lambda function and only use it once.

Reply:
Early in development there was an issue with a single nn.Linear being shared between multiple MLP modules and this was the fix, but I agree its bad practice. I'll radd the change if it passes the convergence tests.